libraries/bot-applicationinsights/src/test/java/com/microsoft/bot/applicationinsights/BotTelemetryClientTests.java

```diff
@@ -0,0 +1,188 @@
+import com.microsoft.applicationinsights.telemetry.RemoteDependencyTelemetry;
+import com.microsoft.applicationinsights.telemetry.PageViewTelemetry;
+import com.microsoft.applicationinsights.telemetry.ExceptionTelemetry;
+import com.microsoft.applicationinsights.telemetry.TraceTelemetry;
+import com.microsoft.bot.builder.BotTelemetryClient;
+import com.microsoft.bot.builder.Severity;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.mockito.Mockito;
+
+import java.time.Duration;
+import java.time.OffsetDateTime;
+import java.util.HashMap;
+import java.util.Map;
+
+public class BotTelemetryClientTests {
+    
+    private BotTelemetryClient botTelemetryClient;
+    private TelemetryChannel mockTelemetryChannel;
+
+    @Before
+    public void initialize() {
+        mockTelemetryChannel = Mockito.mock(TelemetryChannel.class);
+
+        TelemetryConfiguration telemetryConfiguration = new TelemetryConfiguration();
+        telemetryConfiguration.setInstrumentationKey("UNITTEST-INSTRUMENTATION-KEY");
+        telemetryConfiguration.setChannel(mockTelemetryChannel);
+        TelemetryClient telemetryClient = new TelemetryClient(telemetryConfiguration);
+
+        botTelemetryClient = new BotTelemetryClientImpl(telemetryClient);
+    }
+
+    @Test
+    public void nullTelemetryClientThrows() {
+        Assert.assertThrows(IllegalArgumentException.class, () -> { 
+            new BotTelemetryClientImpl(null); 
+        });
+    }
+
+    @Test
+    public void nonNullTelemetryClientSucceeds() {
+        TelemetryClient telemetryClient = new TelemetryClient();
+
+        BotTelemetryClient botTelemetryClient = new BotTelemetryClientImpl(telemetryClient);
+    }
+
+    @Test
+    public void overrideTest() {
+        TelemetryClient telemetryClient = new TelemetryClient();
+        MyBotTelemetryClient botTelemetryClient = new MyBotTelemetryClient(telemetryClient);
+    }
+
+    @Test
+    public void trackAvailabilityTest() {
+        Map<String, String> properties = new HashMap<>();
+        Map<String, Double> metrics = new HashMap<>();
+        properties.put("hello", "value");
+        metrics.put("metric", 0.6);
+
+        botTelemetryClient.trackAvailability(
+            "test",
+            OffsetDateTime.now(),
+            Duration.ofSeconds(1000), // TODO: use computer ticks
+            "run location",
+            true,
+            "message",
+            properties,
+            metrics);
+
+        Mockito.doAnswer(invocation -> {
+            AvailabilityTelemetry availabilityTelemetry = invocation.getArgument(0);
+
+            Assert.assertEquals("test", availabilityTelemetry.getName());
+            Assert.assertEquals("message", availabilityTelemetry.getData().getMessage());
+            Assert.assertEquals("value", availabilityTelemetry.getProperties().get("hello"));
+            Assert.assertEquals((Double) 0.6, availabilityTelemetry.getMetrics().get("metric"));
```

Comment:
Is the `(Double)` necessary? I saw that in other parts you are not casting here. Check if we really need this cast, if we need this, maybe the cast should be in `availabilityTelemetry.getMetrics().get("metric")` instead of `0.6`.

Confirm this

Reply:
see below

Reply:
oh, ok, maybe we could add it to the getMetrics